dm/syncer/dbconn/db.go

```diff
@@ -241,16 +245,41 @@ func (conn *DBConn) ExecuteSQLAutoSplit(
 
 func (conn *DBConn) retryableFn(tctx *tcontext.Context, queries, args any) func(int, error) bool {
 	return func(retryTime int, err error) bool {
-		if retry.IsConnectionError(err) {
-			err = conn.ResetConn(tctx)
+		if retry.IsConnectionError(err) || (conn.cfg.SafeMode && retry.IsRetryableConnectionErrorOnBegin(err)) {
```

Comment:
As discuss in the linked issue https://github.com/pingcap/tiflow/issues/11805#issuecomment-2505470457, I think it's

```suggestion
		if retry.IsConnectionError(err) || (conn.cfg.SafeMode || retry.IsRetryableConnectionErrorOnBegin(err)) {
```

Reply:
Good catch, I changed this to be a `(safemode || begin) && invalidconnection` condition.